cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/ReplicaDistributionGoal.java

```diff
@@ -257,6 +257,8 @@ protected void updateGoalState(ClusterModel clusterModel, Set<String> excludedTo
       LOG.warn("Omitting resource balance limit to relocate remaining replicas from dead brokers/disks.");
       return;
     }
+    // Sanity check: No replica should be move to broker which used to host the replica of same partition on its broken disk.
```

Comment:
Nit: `// Sanity check: No replica should be move to broker which used to host the replica of same partition on its broken disk.` -> `// Sanity check: No replica should be moved to a broker, which used to host any replica of the same partition on its broken disk.`